### PR TITLE
[Improvement][API]optimize tenant code validation

### PR DESCRIPTION
**Pull request by haydenzhourepo**
## Purpose of the pull request

`isNumeric` method can not be used to validate a Linux username.
just add an `isValidLinuxUserName` method to check out if the string is a valid Linux username.
a valid Linux username will match the pattern [a-z_][a-z\d_]{0,30} according to https://stackoverflow.com/questions/6949667/what-are-the-real-rules-for-linux-usernames-on-centos-6-and-rhel-6

this closes #5092 

## Brief change log

  - *Add isValidLinuxUserName to check if tenant cod is valid*

## Verify this pull request

This change added tests and can be verified as follows:
  - *Added isValidLinuxUserName test method in RegexUtils.java to verify the change.*
  - *Manually verified the change by testing locally.*

### CODE REVIEW

**Comment by chengshiwen** [comment]
@haydenzhourepo You need to pass the  **check code style** and the **unit test** at first

**check code style**:
image

**unit test**:
image

**Comment by CalvinKirs** [comment]
You can refer this doc:https://dolphinscheduler.apache.org/zh-cn/community/development/pull-request.html




On 03/18/2021 18:00，Shiwen ***@***.***> wrote：

@haydenzhourepo You need to pass the check code style at first

**Comment by haydenzhourepo** [comment]
@chengshiwen  thanks for your response, I will fix those problems.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/utils/RegexUtils.java [hunk 1]
```diff
@@ -30,6 +30,8 @@
      */
     private static final String CHECK_NUMBER = "^-?\\d+(\\.\\d+)?$";
 
+    private static final String LINUX_USERNAME_PATTERN = "[a-z_][a-z\\d_]{0,30}";
```
Comment (chengshiwen) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/utils/RegexUtils.java:33]: This regular expression pattern is not correct. `[a-zA-Z0-9_.][a-zA-Z0-9_.-]*[$]?` should be used by `man useradd` and it has better compatibility.

image

**Review by chengshiwen — CHANGES_REQUESTED** [review]

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/utils/RegexUtils.java (hunk 1 above)
Comment (haydenzhourepo) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/utils/RegexUtils.java:33]: image

This is the constraints of ubuntu OS above. different Linux OS have different constraints of username.  
We can give the strictest rules to fit all of those different OS. which refers to https://stackoverflow.com/questions/6949667/what-are-the-real-rules-for-linux-usernames-on-centos-6-and-rhel-6

**Comment by chengshiwen** [comment]
This also closes #4383

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/utils/RegexUtils.java (hunk 1 above)
Comment (chengshiwen) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/utils/RegexUtils.java:33]: make sense

**Review by chengshiwen — APPROVED** [review]
+1

**Review by CalvinKirs — APPROVED** [review]
LGTM

**Comment by CalvinKirs** [comment]
considering that it's your first contribution, I think we can get deep communiction, you can contact me through mail or add wechat(Kris_Evil), when mail or added, please tell me who you are, I think I can help to familiar with the DolphinScheduler if you meet with problems.

**Comment by haydenzhourepo** [comment]
OK thanks

**Pull request merged**